### PR TITLE
DOC: optimize: add marginals/slack example to `linprog`

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
gh-11848

#### What does this implement/fix?
@scratchmex wrote:
> Probably worth adding an example with [lagrange multipliers] on the documentation?

Would this have helped you, @scratchmex?

### CODE REVIEW

File: scipy/optimize/_linprog.py [hunk 1]
```diff
@@ -564,6 +564,32 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
     array([10., -3.])
     >>> res.message
     'Optimization terminated successfully. (HiGHS Status 7: Optimal)'
+
+    The marginals (AKA dual values / shadow prices / Lagrange multipliers)
+    and residuals (slacks) are also available.
+    >>> res.ineqlin
```
Comment (mdhaber) [inline scipy/optimize/_linprog.py:570]:
```suggestion

    >>> res.ineqlin
```

**Comment by mdhaber** [comment]
For convenience @scratchmex, here is the rendered documentation.
image
image

and [here is a link](https://output.circle-artifacts.com/output/job/b985d415-1dd2-4693-845a-5646d61080d2/artifacts/0/html-scipyorg/reference/generated/scipy.optimize.linprog.html#scipy.optimize.linprog).

**Comment by mdhaber** [comment]
@bharat29

**Comment by EwoutH** [comment]
Thanks for this effort, any chance it can still make the 1.9.0 release?

**Comment by mdhaber** [comment]
We don't typically backport documentation enhancements, but it is not impossible. Do you think it's important?
If so, can you review for accuracy, and is it helpful?

**Comment by EwoutH** [comment]
Thanks for you very quick reply! I find it very useful, especially since the `ineqlin` function isn't mentioned anywhere else on the page, and also not at the [scipy.optimize.OptimizeResult](https://output.circle-artifacts.com/output/job/b985d415-1dd2-4693-845a-5646d61080d2/artifacts/0/html-scipyorg/reference/generated/scipy.optimize.OptimizeResult.html#scipy.optimize.OptimizeResult) page.

What's also a bit confusing is that the `highs` examples are over the main `scipy.optimize.linprog` page, while the specific parameters and returns are at `linprog(method=’highs’)`. There is no single page where you can read everything and search (ctrl + f) through.

> We don't typically backport documentation enhancements, but it is not impossible. Do you think it's important?

Most users will read the docs of the stable version, which is very logical, since that's probably the version you're using. If we have to wait to 1.10 for this example to land, so many users will have missed it. So in my opinion all examples and clarifications should be backported to the current stable branch, since that's what users use.

> If so, can you review for accuracy?

Unfortunately I'm still learning linear programming and not at a level yet where I can determine the accuracy of this exact example.

**Comment by mdhaber** [comment]
> There is no single page where you can read everything and search (ctrl + f) through.

I suppose that's part of why we are deprecating the old methods that don't have these capabilities. Once we do, everything would be on the main page.

However, i suppose at this point we could move the HiGHS stuff to the main `linprog` page and just note that it is only available with `method='highs'`. Would that be good?

**Comment by EwoutH** [comment]
> However, i suppose at this point we could move the HiGHS stuff to the main linprog page and just note that it is only available with method='highs'. Would that be good?

Yes, I think that's a good solution, especially since HiGHS is the default.

**Review by mckib2 — APPROVED** [review]
Nice little example -- LGTM!  This is only a docstring change that has passed before, so I don't think we need to waste the CI resources on another run, I'll go ahead and merge

**Pull request merged**